packages/listr2/src/utils/concurrency.ts

```diff
@@ -24,6 +29,14 @@ export class Concurrency {
     })
   }
 
+  private initPromise (): void {
+    const { promise, resolve, reject } = this.withResolvers<void>()
+
+    this.promise = promise
+    this.resolve = resolve
+    this.reject = reject
+  }
```

Comment:
If we are going this route, why not just initiate it on the constructor and offload it to a function. Same for the `withResolvers`.